Blocks/HTTP/test.cc

```diff
@@ -985,96 +1057,195 @@ TEST(HTTPAPI, ServeStaticFilesFrom) {
+  // so after redirects, the response code and body are always HTTP 200 with "Done.", not the targeted content.
+  const auto scope_redirect_to = HTTP(FLAGS_net_api_test_port_secondary)
+                                     .Register("/", URLPathArgs::CountMask::Any, [](Request r) { r("Done."); });
+
+  const auto scope = HTTP(FLAGS_net_api_test_port).ServeStaticFilesFrom(dir,
+      ServeStaticFilesFromOptions{"/static/something", Printf("http://localhost:%d/anything", FLAGS_net_api_test_port_secondary)});
+
+  // Root index file.
+  EXPECT_EQ("<h1>HTML index</h1>", HTTP(GET(Printf("http://localhost:%d/static/something/", FLAGS_net_api_test_port))).body);
+
+  // Redirect from directory without trailing slash to directory with trailing slash.
+  {
+    ASSERT_THROW(HTTP(GET(Printf("http://localhost:%d/static/something", FLAGS_net_api_test_port))),
+                 HTTPRedirectNotAllowedException);
+    const auto dir_response = HTTP(GET(Printf("http://localhost:%d/static/something", FLAGS_net_api_test_port)).AllowRedirects());
+    EXPECT_EQ(200, static_cast<int>(dir_response.code));
+    EXPECT_EQ(Printf("http://localhost:%d/anything/", FLAGS_net_api_test_port_secondary), dir_response.url);
+    EXPECT_EQ("Done.", dir_response.body);
+  }
+
+  // Subdirectory index file.
+  EXPECT_EQ("<h1>HTML sub_dir index</h1>", HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir/", FLAGS_net_api_test_port))).body);
+
+  // Redirect from subdirectory without trailing slash to subdirectory with trailing slash.
+  {
+    ASSERT_THROW(HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir", FLAGS_net_api_test_port))),
+                 HTTPRedirectNotAllowedException);
+    const auto sub_dir_response = HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir", FLAGS_net_api_test_port)).AllowRedirects());
+    EXPECT_EQ(200, static_cast<int>(sub_dir_response.code));
+    EXPECT_EQ(Printf("http://localhost:%d/anything/sub_dir/", FLAGS_net_api_test_port_secondary), sub_dir_response.url);
+    EXPECT_EQ("Done.", sub_dir_response.body);
+  }
+
+  // Subsubdirectory index file.
+  EXPECT_EQ("<h1>HTML sub_sub_dir index</h1>", HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir/sub_sub_dir/", FLAGS_net_api_test_port))).body);
+
+  // Redirect from subsubdirectory without trailing slash to subsubdirectory with trailing slash.
+  {
+    ASSERT_THROW(HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir/sub_sub_dir", FLAGS_net_api_test_port))),
+                 HTTPRedirectNotAllowedException);
+    const auto sub_sub_dir_response = HTTP(GET(Printf("http://localhost:%d/static/something/sub_dir/sub_sub_dir", FLAGS_net_api_test_port)).AllowRedirects());
+    EXPECT_EQ(200, static_cast<int>(sub_sub_dir_response.code));
+    EXPECT_EQ(Printf("http://localhost:%d/anything/sub_dir/sub_sub_dir/", FLAGS_net_api_test_port_secondary), sub_sub_dir_response.url);
+    EXPECT_EQ("Done.", sub_sub_dir_response.body);
+  }
```

Comment:
What's the issue with the long lines? Why should I terminate a line at a point where it hasn't yet said what it had to say? Broken lines lead to harder merge conflict resolution because they get mixed up by the source control diffing algorithms that are line-based. This specific C++ coding style where the lines break at different logical points does not help to minimize the diffs and avoid/simplify merge conflict resolution.